src/libraries/System.Formats.Tar/tests/TarReader/TarReader.GetNextEntry.Tests.cs

```diff
@@ -250,5 +250,43 @@ public void GetNextEntry_UnseekableArchive_ReplaceDataStream_ExcludeFromDisposin
                 Assert.Equal("Substituted", streamReader.ReadLine());
             }
         }
+
+        [Theory]
+        [InlineData(512)]
+        [InlineData(512 + 1)]
+        [InlineData(512 + 512 - 1)]
+        public void BlockAlignmentPadding_DoesNotAffectNextEntries(int contentSize)
+        {
+            byte[] fileContents = new byte[contentSize];
+            Array.Fill<byte>(fileContents, 0x1);
+
+            using var archive = new MemoryStream();
+            using (var writer = new TarWriter(archive, leaveOpen: true))
+            {
+                var entry1 = new PaxTarEntry(TarEntryType.RegularFile, "file");
+                entry1.DataStream = new MemoryStream(fileContents);
+                writer.WriteEntry(entry1);
+
+                var entry2 = new PaxTarEntry(TarEntryType.RegularFile, "next-file");
+                writer.WriteEntry(entry2);
+            }
+
+            archive.Position = 0;
+            using var unseekable = new WrappedStream(archive, archive.CanRead, archive.CanWrite, canSeek: false);
+            using var reader = new TarReader(unseekable);
+
+            TarEntry e = reader.GetNextEntry();
+            Assert.Equal(contentSize, e.Length);
```

Comment:
I would've added a couple other asserts when checking both entries:
```cs
Assert.Equal("file", e.Name);
Assert.Equal(TarEntryType.RegularFile, e.EntryType);
```
The `name` field starts at byte 0 of the header. But I guess `GetNextEntry()` would throw if we aren't reading from the right position when it encounters an unexpected character.